### PR TITLE
samples: matter: Reduce memory usage

### PR DESCRIPTION
**Pull request by adigie**
- Remove custom heap size from nRF54L15 configs.
- Reduce Packet Buffer Pool size to 11.

### CODE REVIEW

File: samples/matter/light_bulb/boards/nrf54l15dk_nrf54l15_cpuapp.conf
Comment (LuDuda) [inline samples/matter/light_bulb/boards/nrf54l15dk_nrf54l15_cpuapp.conf:11]: I wonder if we can remove this line completely and have a default values in sdk-connectedhomeip only (potentially different for nRF52/53/54).

Btw. In this PR you have changes configuration only for nRF54L15 - what about 52 and 53?

Also i believe we should run nightly on this PR before merging - as previously some of the Matter tests were failing (maybe its not the case anymore): https://github.com/nrfconnect/sdk-nrf/commit/cdcb5801a946e9969780f9d7b8d935654751c77e

Comment (adigie) [reply samples/matter/light_bulb/boards/nrf54l15dk_nrf54l15_cpuapp.conf]: Yes, we can. We will use default configuration for all platforms - 8KB.

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/memory_optimization.rst:423]:
```suggestion
The Matter application uses a fixed-size pool for packet buffer memory management.
```

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/memory_optimization.rst:425]:
```suggestion
You can adjust the pool size to reduce the memory usage of your application, but that can have a negative effect on the performance of the application.
```

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (wiba-nordic) [inline doc/nrf/protocols/matter/getting_started/memory_optimization.rst:429]: Is it intentional that this note is under the new heading instead of the previous one?

**Review by wiba-nordic — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (adigie) [reply doc/nrf/protocols/matter/getting_started/memory_optimization.rst:429]: Yes, this note refers to configuration of pool used in Thread, so it fits better here.

**Comment by wiba-nordic** [comment]
I'll hold on approving for a bit since I'm also part of ncs-matter, so one of the guys can properly approve.

**Comment by adigie** [comment]
@nrfconnect/ncs-code-owners Please review.

**Review by gmarull — APPROVED** [review]

**Comment by ArekBalysNordic** [comment]
@peknis could you please look at it? ncs-matter-doc code owner has been changed and we need a new approval.

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/protocols/matter/getting_started/memory_optimization.rst
Comment (adigie) [reply doc/nrf/protocols/matter/getting_started/memory_optimization.rst]: Fixed, thanks!

**Review by peknis — APPROVED** [review]

**Comment by adigie** [comment]
@ArekBalysNordic we can merge this PR, as @peknis approved it and then I re-requested a review by mistake.

**Pull request merged**
